src/common/transactions/transaction-utils.ts

```diff
@@ -119,3 +130,84 @@ export const generateSTXTransferTx = ({
 
 export const stacksTransactionToHex = (transaction: StacksTransaction) =>
   `0x${transaction.serialize().toString('hex')}`;
+
+export const isContractTokenTransfer = (tx: Tx) => {
+  return tx.tx_type === 'contract_call' && tx.contract_call.function_name === 'transfer';
+};
+
+// calculate the real amount of the token based on the decimal number
+// specified in the corresponding token smart contract
+export const tokenRealAmount = (tx: ContractCallTransaction) => {
+  if (!tx.contract_call?.function_args) return;
+  const val = tx.contract_call.function_args[0].repr.replace('u', '');
+  const x = new BigNumber(val);
+  // get the decimal number for this token
+  const asset = assetById(tx.contract_call.contract_id);
+  if (!asset || !asset.meta) return;
+  const y = new BigNumber(asset.meta.decimals);
+  return x.dividedBy(10 ** y.toNumber());
```

Comment:
This is doing maths outside of big number, probably need to use `exponentiatedBy` or `.pow`